include/Command.h

```diff
@@ -0,0 +1,158 @@
+            if (str[start] == '$' && end > start + 1) {
+                std::string varName = str.substr(start + 1, end - start - 1);
+                char* varValue = getenv(varName.c_str());
+                if (varValue != nullptr) {
+                    result += varValue;
+                }
+                start = end + 1;
+            }
+            else if (str[start] == '{' && end > start + 1 && str[end - 1] == '$') {
+                std::string varName = str.substr(start + 2, end - start - 2);
+                char* varValue = getenv(varName.c_str());
+                if (varValue != nullptr) {
+                    result += varValue;
+                }
+                start = end + 1;
+            }
+            else {
+                result += str.substr(start, end - start + 1);
+                start = end + 1;
+            }
+        }
+        if (start != str.length()) {
+            result += str.substr(start);
+        }
+        return result;
+    }
+
+};
+
+class EchoCommand : public Command {
+private:
+    std::vector<std::string> args;
+public:
+    EchoCommand(const std::vector<std::string>& args) : args(args) {}
+    void execute() override {
+        for (size_t i = 1; i < args.size(); ++i) {
+            std::string expandedArg = expandVariables(args[i]);
+            std::cout << expandedArg << " ";
+        }
+        std::cout << std::endl;
+    }
+};
+
+class ExternalCommand : public Command {
+private:
+    std::vector<std::string> args;
+    bool runInBackground;
+public:
+    ExternalCommand(const std::vector<std::string>& args, bool runInBackground) : args(args), runInBackground(runInBackground) {}
+    void execute() override {
+        // Convert argument vector to array of C-style strings
+        std::vector<char*> argv;
+        for (const auto& arg : args) {
+            argv.push_back(const_cast<char*>(arg.c_str()));
+        }
+        argv.push_back(nullptr);
+
+        // Create child process
+        pid_t pid = fork();
+        if (pid == -1) {
+            throw std::runtime_error(CHILD_CREATION_FAILED);
+        }
+        else if (pid == 0) {
+            // Child process
+            if (execvp(argv[0], argv.data()) == -1) {
+                // Print the value of the variable if it exists
+                char* varValue = getenv(args[0].c_str());
+                if (varValue != nullptr) {
+                    std::cout << varValue << std::endl;
+                }
+                else {
+                    std::string command = args[0];
+                    if (command[0] == '$') {
+                        command = command.substr(1);
+                    }
+                    std::vector<std::string> newArgs = { "/bin/sh", "-c", command };
+                    ExternalCommand(newArgs, runInBackground).execute();
+                }
+            }
+        }
+        else {
+            // Parent process
+            if (!runInBackground) {
```

Comment:
What if it IS running in background? You still need a way to be able to track your background jobs